create and delete all account integrations

Varsha-Arun opened the pull request:
Creating and deleting all account integrations. Completed account integration test suite in 47secs

![selection_999 1301](https://user-images.githubusercontent.com/15084465/33977009-b312ccb2-e0be-11e7-9c92-f68f5278b721.png)

![selection_999 1302](https://user-images.githubusercontent.com/15084465/33977016-b898e34c-e0be-11e7-908e-76e76873b80e.png)

Varsha-Arun commented:
![selection_999 1323](https://user-images.githubusercontent.com/15084465/34285987-d2015a44-e703-11e7-9949-a00f6fa4184c.png)

latest PR(https://github.com/Shippable/bvt/pull/2/commits/e65df7a09f1fca4f6703d1c179154fdfcc5e9f4a)  testing subscription API calls for 
1. GET subscriptions for owner, member, collaborator and public user
2. GET subscription by ID for owner, member, collaborator, unauthorized and public user
3. GET run status by subscription ID for owner, member, collaborator, unauthorized and public user

avinci commented on tests/core/api/GH_ORG_SUBS_API.js:
```diff
@@ -0,0 +1,436 @@
+'use strict';
```
the only issue i have here is the name of the file. its not GH_ORG...its just subscriptions.js right? The api does not give you different results when you use different source control. by qualifying this by GH_ORG you are hinting that we should do a bitbucket test also which is pointless.. All we are testing here is the subs and making sure the subs are returned correctly.

avinci commented on tests/core/api/GH_ORG_SUBS_API.js:
```diff
@@ -0,0 +1,436 @@
+
+    it('2. Owner can get subscription by ID',
+      function (done) {
+        var sub = _.findWhere(subscriptions, {orgName:global.TEST_GH_ORGNAME}) || {};
```
why do we need to do this? why not just use the project that you setup, use its subscriptionId

avinci commented on tests/core/api/GH_ORG_SUBS_API.js:
```diff
@@ -0,0 +1,436 @@
+
+    it('3. Member can get subscription by ID',
+      function (done) {
+        var sub = _.findWhere(subscriptions, {orgName:global.TEST_GH_ORGNAME}) || {};
```
same issue here, we should use the same project and make sure all the folks can get it.

avinci commented on tests/core/api/GH_ORG_SUBS_API.js:
```diff
@@ -0,0 +1,436 @@
+
+    it('5. Public user cannot get subscription by ID',
+      function (done) {
+        var sub = _.findWhere(subscriptions, {orgName:global.TEST_GH_ORGNAME}) || {};
```
we are missing one more test here. We should that public users cannot get any values if the make a getS call without Ids. We should also test getS call by passing a subscriptionId

avinci commented on tests/core/api/GH_ORG_SUBS_API.js:
```diff
@@ -0,0 +1,436 @@
+      }
+    );
+
+    it('16. public user cannot get Run Status by subscription ID',
```
we also need to test the public project where the unauthorized and public guys can get this information correct?

Varsha-Arun commented:
Updated the PR
Added tests for GET build status from subscriptions/{subscriptionId}
And whether it is private or public project response for getting latest runStatus through subscription ID is same for unauthorized and public user.

Varsha-Arun commented:
Adding project API call tests

![selection_999 1333](https://user-images.githubusercontent.com/15084465/34673779-7a77bd50-f4a9-11e7-969e-cc3d509440fc.png)

- GET projects
- GET project by ID
- POST triggerNewBuild by project ID
